script/tool/test/pubspec_check_command_test.dart

```diff
@@ -177,12 +181,19 @@ ${dependenciesSection()}
 ${devDependenciesSection()}
 ''');
 
-      final Future<List<String>> result =
-          runCapturingPrint(runner, <String>['pubspec-check']);
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['pubspec-check'], errorHandler: (Error e) {
+        commandError = e;
+      });
 
-      await expectLater(
-        result,
-        throwsA(isA<ToolExit>()),
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains(
+              'Found a "homepage" entry; only "repository" should be used.'),
+        ]),
```

Comment:
These changes are opportunistic cleanup; I noticed when duplicating tests as a starting point that when I wrote these tests I copied from what I have since discovered is an anti-pattern: not checking the details of the output on failures, which means failure tests can pass for the wrong reasons. This `commandError`+output pattern is what I've been applying throughout the codebase as I find the older style.